enhancements/sig-architecture/32-extensiblescheduling/32-extensiblescheduling.md

```diff
@@ -0,0 +1,321 @@
+// as a floating betwween 0 and 1.
+type Prioritizer interface {
+	Plugin
+
+	// PreScore() do some prepare work before Score().
+  // For example, creating ManagedClusterScalar CR for "customized-" prefixed prioritizer.
+	PreScore(ctx context.Context, placement *clusterapiv1alpha1.Placement, clusters []*clusterapiv1.ManagedCluster) error
+
+	// Score gives the score to a list of the clusters,  it returns a map with the key as
+	// the cluster name.
+	Score(ctx context.Context, placement *clusterapiv1alpha1.Placement, clusters []*clusterapiv1.ManagedCluster) (map[string]int64, error)
+}
+```
+
+Add GC() funcion in placement controller, to clean not used ManagedClusterScalar CR every 10 minutes.
+
+### 3. Let prioritizer plugin support schedule with customized value in ManagedClusterScalar.
+Add a new plugin called customized plugin, which can score clusters based on ManagedClusterScalar CR `customized-{prioritizer-name}`.
+
+### 4. Workflow
+Refer to [Examples](#Examples)
+
+### 5. Open Questions
+1. Is one score enough? any other missed use cases? For example, spreading depends on previous choice.
+
+## Alternative
+Using scheduler extender mechanism, to support schedule based on metrics, schedule based on SLA.
+
+Pros: 
+  - More flexible compared with above proposal can only store one score.
+
+Cons:
+  - Calling extender by http/https may have performance issue.
+
+## Examples
+
+### 1. An agent on managed cluster update ResourceRatioCPU score to API, and placement prioritizer plugin score based on it. (Use Story 1)
+![](./pic1.png)
+1. User create a new placement as below
+```yaml
+apiVersion: cluster.open-cluster-management.io/v1alpha1
+kind: Placement
+metadata:
+  name: placement
+  namespace: ns1
+spec:
+  numberOfClusters: 1
+  prioritizerPolicy:
+    mode: Exact
+    configurations:
+      - name: Customized-ResourceRatioCPU
+```
+2. Placement-controller watch the placement changes and generate a new ManagedClusterScalar CR `customized-resourceratiocpu` to trigger customized data collection.
+```yaml
+apiVersion: cluster.open-cluster-management.io/v1alpha1
+kind: ManagedClusterScalar
+metadata:
+  name: customized-resourceratiocpu
+  namespace: cluster1
+```
+3. The agent on managed cluster watch the new ManagedClusterScalar CR and update score into it. The agent on managed cluster prerefresh the score periodically before expire time.
+```yaml
+apiVersion: cluster.open-cluster-management.io/v1alpha1
+kind: ManagedClusterScalar
+metadata:
+  name: customized-resourceratiocpu
+  namespace: cluster1
+status:
+  conditions:
+  - lastTransitionTime: "2021-10-28T08:31:39Z"
+    message: ManagedClusterScalar updated successfully
+    reason: ManagedClusterScalarUpdated
+    status: "True"
+    type: ManagedClusterScalarUpdated
+  expireTime: "2021-10-29T18:31:39Z"
+  score: 74
+```
+4. Placement read value from CR and score on it. (when to trigger re-schedule???)
```

Comment:
different clusters update this value separately. So how can the scheduler to know it has sufficient data to schedule?